### PR TITLE
Remove nuget.exe dependency for DotNetCoreCLI

### PR DESCRIPTION
**Pull request by aldoms**

### CODE REVIEW

File: Tasks/Common/nuget-task-common/NuGetXmlHelper.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import * as tl from "vsts-task-lib";
+
+import { INuGetXmlHelper } from "./INuGetXmlHelper";
+
+export class NuGetXmlHelper implements INuGetXmlHelper {
+    constructor(private _nugetConfigPath: string) {
+    }
+
+    public SetApiKeyInNuGetConfig(source: string, apiKey: string): void {
+        tl.debug('Setting ApiKey is not supported without using nuget.exe');
```
Comment (pspill) [inline Tasks/Common/nuget-task-common/NuGetXmlHelper.ts]: You mentioned earlier this should probably be higher priority message than debug.

File: Tasks/Common/nuget-task-common/NuGetXmlHelper.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    public AddSourceToNuGetConfig(name: string, source: string, username?: string, password?: string): void {
+        NuGetXmlHelper._updateXmlFile(this._nugetConfigPath, (xml: any): any => {
+            if (xml) {
+                let xmlPackageSources = NuGetXmlHelper._getLastElement(xml, "packageSources");
```
Comment (pspill) [inline Tasks/Common/nuget-task-common/NuGetXmlHelper.ts]: I was surprised at first that _getLastElement will create the element if it is not found.  Either we should change the function name to reflect this, or separate out _getLastElement and _createElement.

File: Tasks/Common/nuget-task-common/NuGetXmlHelper.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        }
+
+        return xml;
+    }
+
+    private static _getLastElement(xml: any, elementName: string): any {
+        if (xml) {
+            let xmlElements = xml.getChildren(elementName);
+            if (!xmlElements || xmlElements.length === 0) {
+                xmlElements = [xml.c(elementName)];
```
Comment (pspill) [inline Tasks/Common/nuget-task-common/NuGetXmlHelper.ts]: Was a bit confused here, until I realized that xml.c will insert this new element just inside the root, which is the correct place for "packageSources" and "packageSourceCredentials".
Maybe a comment here, maybe we should throw if the element name doesn't match one of these two to make it clear what we can legally insert.

File: Tasks/Common/nuget-task-common/NuGetXmlHelper.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
+import * as fs from "fs";
```
Comment (pspill) [inline Tasks/Common/nuget-task-common/NuGetXmlHelper.ts]: Can we write a unit test that just targets NuGetXmlHelper?

File: Tasks/Common/nuget-task-common/package.json [hunk 5]
```diff
@@ -13,9 +13,11 @@
   },
   "license": "MIT",
   "dependencies": {
+    "ltx": "^2.6.2",
```
Comment (pspill) [inline Tasks/Common/nuget-task-common/package.json]: do we need oss approval for ltx?

File: Tasks/DotNetCoreCLI/pushcommand.ts [hunk 6]
```diff
@@ -79,7 +79,7 @@ export async function run(): Promise<void> {
         const tempNuGetPath = path.join(tempNuGetConfigDirectory, "nuget.config");
         tl.mkdirP(tempNuGetConfigDirectory);
 
-        const nuGetConfigHelper = new NuGetConfigHelper2(nuGetPath, null, authInfo, { credProviderFolder: null, extensionsDisabled: true }, tempNuGetPath);
+        const nuGetConfigHelper = new NuGetConfigHelper2(nuGetPath, null, authInfo, { credProviderFolder: null, extensionsDisabled: true }, tempNuGetPath, false);
```
Comment (pspill) [inline Tasks/DotNetCoreCLI/pushcommand.ts]: add a comment or local that clarifies what null and false signify.

File: Tasks/DotNetCoreCLI/restorecommand.ts [hunk 7]
```diff
@@ -74,7 +74,8 @@ export async function run(): Promise<void> {
             nuGetConfigPath,
             authInfo,
             { credProviderFolder: null, extensionsDisabled: true },
-            null);
+            null,
+            false);
```
Comment (pspill) [inline Tasks/DotNetCoreCLI/restorecommand.ts]: same here

File: Tasks/Common/nuget-task-common/NuGetXmlHelper.ts (hunk 4 above)
Comment (aldoms) [inline Tasks/Common/nuget-task-common/NuGetXmlHelper.ts]: Good idea, I've added a test to validate NuGetXmlHelper adds the xml elements as expected.

File: Tasks/Common/nuget-task-common/package.json (hunk 5 above)
Comment (aldoms) [inline Tasks/Common/nuget-task-common/package.json]: No, it's already being used in other tasks and it's already approved.

**Pull request merged**